support/mender-client.service

```diff
@@ -1,6 +1,7 @@
 [Unit]
 Description=Mender OTA update service
-After=systemd-resolved.service
+Wants=network-online.target
+After=systemd-resolved.service network-online.target
```

Comment:
I think using `After` should be enough here, no? I would prefer not to use `Wants` if possible to avoid adding dependencies if people have special network setups that don't use `network-online.target`.

Reply:
The reason this is `Wants` and not `Requires` is exactly so that network-online.target is optional.

This change is meant to provide a sane default. Without the `Wants` here, the network-online.target would only be started by accident unless the user knew they had to include it in some other way; none of the standard Yocto targets include network-online.target directly, it's pulled in by the services that need it (such as autofs.service, where it's also `Wants`).

For the special cases, the user would be doing something knowingly. There, the user would purposefully not include the network-online.target or not put anything in it, and both would work fine.

Reply:
Alright, you convinced me! :-)

I will start our full test suite.